### PR TITLE
dnos9_command (new module)

### PR DESCRIPTION
**Pull request by skg-net**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME
- modules/core/network/dnos9/dnos9_command.py
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel 6744332600) last updated 2016/08/31 14:21:15 (GMT -700)
  lib/ansible/modules/core: (detached HEAD 86366eca68) last updated 2016/08/31 14:21:24 (GMT -700)
  lib/ansible/modules/extras: (detached HEAD aedaca55ac) last updated 2016/08/31 14:21:24 (GMT -700)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

Addded support for dnos9_command module.

### CODE REVIEW

**Comment by skg-net** [comment]
Hi @gundalow @privateip, Please take a look it.

**Comment by skg-net** [comment]
Reason for build failure is the dependent PR https://github.com/ansible/ansible/pull/17468 not merged yet.

File: lib/ansible/utils/module_docs_fragments/dnos9.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    required: true
+  port:
+    description:
+      - Specifies the port to use when building the connection to the remote
+        device.
+    required: false
+    default: 22
+  username:
+    description:
+      - This value I(username) is used to authenticate the SSH session to the
```
Comment (gundalow) [inline lib/ansible/utils/module_docs_fragments/dnos9.py]: You don't need to repeat the option name with I() when describing that option.

"User to authenticate..."

File: lib/ansible/utils/module_docs_fragments/dnos9.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        remote device. If the value is not specified in the task, the
+        value of environment variable ANSIBLE_NET_USERNAME will be used instead.
+    required: false
+  password:
+    description:
+      - This value I(password) is used to authenticate the SSH session to
```
Comment (gundalow) [inline lib/ansible/utils/module_docs_fragments/dnos9.py]: "Password to authenticate the SSH session to..."

File: lib/ansible/utils/module_docs_fragments/dnos9.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  password:
+    description:
+      - This value I(password) is used to authenticate the SSH session to 
+        the remote device. If the value is not specified in the task, the
+        value of environment variable ANSIBLE_NET_PASSWORD will be used instead.
+    required: false
+    default: null
+  ssh_keyfile:
+    description:
+      - This value I(ssh_keyfile) is the path to the key used to authenticate
```
Comment (gundalow) [inline lib/ansible/utils/module_docs_fragments/dnos9.py]: "Path to an ssh key used to authenticate..."

**Comment by gundalow** [comment]
needs_revision

**Comment by gundalow** [comment]
Where is `auth_pass` defined, can you please confirm that it had `no_log` set so we don't end up displaying passwords to the screen.

**Comment by skg-net** [comment]
Its defined in the modules/network.py, the no_log is set to True. We use it in the authorize function.

**Comment by skg-net** [comment]
ready_for_review

File: CHANGELOG.md [hunk 4]
```diff
@@ -46,6 +46,8 @@ Ansible Changes By Release
   * dnos10_command
   * dnos10_config
   * dnos10_template
+- dnos9
```
Comment (gundalow) [inline CHANGELOG.md]: dnos9 should come before dnos10.

**Comment by gundalow** [comment]
Rebase needed

**Comment by privateip** [comment]
Looks like you have done a merge commit and not a rebase.  Please fix the PR and remove the merge commit before we can accept this PR

**Comment by skg-net** [comment]
@privateip @gundalow. I have completed the rebase, please take a look at it.
ready_for_review

**Comment by privateip** [comment]
needs_rebase

**Comment by skg-net** [comment]
@privateip @gundalow. I have completed the rebase again, please take a look at it.
ready_for_review

**Pull request merged**